mindee/client.py

```diff
@@ -0,0 +1,318 @@
+                )
+        return self._make_request(doc_config, include_words)
+
+    def _make_request(self, doc_config: DocumentConfig, include_words: bool):
+        if doc_config.api_type == API_TYPE_OFF_THE_SHELF:
+            response = doc_config.constructor.request(
+                doc_config.endpoints, self.input_doc, include_words=include_words
+            )
+        else:
+            response = doc_config.constructor.request(
+                doc_config.endpoints, self.input_doc
+            )
+
+        dict_response = response.json()
+
+        if response.status_code > 201 and self.raise_on_error:
+            raise HTTPException(
+                "API %s HTTP error: %s"
+                % (response.status_code, json.dumps(dict_response))
+            )
+        if response.status_code > 201:
+            return Response(
+                doc_config,
+                http_response=dict_response,
+                pages=[],
+                document_type=doc_config.document_type,
+                document=None,
+            )
+        return format_response(
+            doc_config, dict_response, doc_config.document_type, self.input_doc
+        )
+
+
+class Client:
+    _doc_configs: DocumentConfigDict
+    raise_on_error: bool = True
+
+    def __init__(self, raise_on_error: bool = True):
+        """
+        :param raise_on_error: Raise an Exception on HTTP errors
+        """
+        self._doc_configs = {
+            ("mindee", "receipt"): Receipt.get_document_config(),
+            ("mindee", "invoice"): Invoice.get_document_config(),
+            ("mindee", "financial_doc"): FinancialDocument.get_document_config(),
+            ("mindee", "passport"): Passport.get_document_config(),
+        }
+        self.raise_on_error = raise_on_error
+
+    def config_custom_doc(
+        self,
+        document_type: str,
+        singular_name: str,
+        plural_name: str,
+        username: str,
+        api_key: str = "",
+    ):
+        """
+        Configure a custom document using the Mindee API Builder.
+
+        See: https://developers.mindee.com/docs/
+
+        :param document_type: The "document type" field in the "Settings" page of the API Builder
+        :param singular_name: The name of the attribute used to retrieve a *single* document from the API response
+        :param plural_name: The name of the attribute used to retrieve *multiple* documents from the API response
+        :param username: Your organization's username on the API Builder
+        :param api_key: Your API key for the endpoint
+        """
+        self._doc_configs[(username, document_type)] = DocumentConfig(
+            {
+                "document_type": document_type,
+                "singular_name": singular_name,
+                "plural_name": plural_name,
+                "username": username,
+                "api_key": api_key,
+            }
+        )
+        return self
+
+    def config_invoice(self, api_key: str = None):
```

Comment:
Given the cascading pattern here, would it be better to call all the methods `Client.add_xxxxx` instead of `Client.config_xxxxx` ?

Reply:
I don't agree with this one. 

I think `config_xxx` is more explicit for a new user than `add_xxx`.